### ISSUE TITLE
Allow pass criteria array to `Group`  (reader) filter constructor

### ISSUE THREAD

**Issue by vjik** [yiisoft/data#123]
Now we can:
- pass filters to constructor
- use `withCriteriaArray()` for set filters or criteria array.

Suggest allow pass to constructor mixed array of filters and criteria (as in `withCriteriaArray()`) and remove `withCriteriaArray()`.

**Comment by samdark** [comment]
I don't think it is a good idea since usually use-cases are different. Criteria array input source is usually user input or database (any it does not accept filter instances) while input source for filter instance is pure code.
